R/run_pacta_reporting_process.R

```diff
@@ -0,0 +1,317 @@
+  survey_dir = Sys.getenv("SURVEY_DIR"),
+  score_card_dir = Sys.getenv("SCORE_CARD_DIR")
+) {
+  log_debug("Checking configuration.")
+  if (is.null(analysis_output_dir) || analysis_output_dir == "") {
+    log_error("ANALYSIS_OUTPUTS_DIR not set.")
+    stop("ANALYSIS_OUTPUTS_DIR not set.")
+  }
+  if (is.null(benchmarks_dir) || benchmarks_dir == "") {
+    log_error("BENCHMARKS_DIR not set.")
+    stop("BENCHMARKS_DIR not set.")
+  }
+  if (is.null(report_output_dir) || report_output_dir == "") {
+    log_error("REPORT_OUTPUT_DIR not set.")
+    stop("REPORT_OUTPUT_DIR not set.")
+  }
+  if (is.null(summary_output_dir) || summary_output_dir == "") {
+    log_error("SUMMARY_OUTPUT_DIR not set.")
+    stop("SUMMARY_OUTPUT_DIR not set.")
+  }
+
+  # defaulting to WARN to maintain current (silent) behavior.
+  logger::log_threshold(Sys.getenv("LOG_LEVEL", "INFO"))
+  logger::log_formatter(logger::formatter_glue)
+
+  # -------------------------------------------------------------------------
+
+  log_info("Starting portfolio report process")
+
+  # Read Params
+  log_trace("Processing input parameters.")
+  if (length(raw_params) == 0L || all(raw_params == "")) {
+    log_error("No parameters specified.")
+  }
+
+  # log_trace("Validating raw input parameters.")
+  # raw_input_validation_results <- jsonvalidate::json_validate(
+  #   json = raw_params,
+  #   schema = system.file(
+  #     "extdata", "schema", "rawParameters.json",
+  #     package = "workflow.pacta.report"
+  #   ),
+  #   verbose = TRUE,
+  #   greedy = FALSE,
+  #   engine = "ajv"
+  # )
+  # if (raw_input_validation_results) {
+  #   log_trace("Raw input parameters are valid.")
+  # } else {
+  #   log_error(
+  #     "Invalid raw input parameters. ",
+  #     "Must include \"inherit\" key, or match full schema."
+  #   )
+  #   stop("Invalid raw input parameters.")
+  # }
+
+  params <- pacta.workflow.utils:::parse_params(
+    json = raw_params,
+    inheritence_search_paths = system.file(
+      "extdata", "parameters",
+      package = "workflow.pacta.report"
+    ) ,
+    schema_file = NULL
+    # schema_file = system.file(
+    #   "extdata", "schema", "portfolioParameters_0-0-1.json",
+    #   package = "workflow.pacta.report"
+    # )
+  )
+
+  # quit if there's no relevant PACTA assets -------------------------------------
+
+  log_debug("Checking for PACTA relevant data in portfolio results.")
+  total_portfolio_path <- file.path(analysis_output_dir, "total_portfolio.rds")
+  if (file.exists(total_portfolio_path)) {
+    total_portfolio <- readRDS(total_portfolio_path)
+    log_trace("Checking for PACTA relevant data in file: \"{total_portfolio_path}\".")
+    pacta.portfolio.utils::quit_if_no_pacta_relevant_data(total_portfolio)
+  } else {
+    log_warn("file \"{total_portfolio_path}\" does not exist.")
+    warning("This is weird... the `total_portfolio.rds` file does not exist in the `30_Processed_inputs` directory.")
```

Comment:
nit: does `30_Processed_inputs` still exist in this context?

Reply:
It does not.